templates/definition/meter/kostal-plenticore.yaml

```diff
@@ -75,16 +75,46 @@ render: |
     source: sunspec
     {{- include "modbus" . | indent 2 }}
     value: 802:SoC # 802 battery control
-  limitsoc:
+  batterymode:
     source: watchdog
     timeout: {{ .watchdog }} # re-write at timeout/2
+    reset: 1 # reset watchdog on normal
     set:
-      source: modbus
-      {{- include "modbus" . | indent 4 }}
-      register:
-        address: 1042 # limit soc
-        type: writemultiple
-        encoding: {{ if (eq .endianness "big") }}float32{{ else }}float32s{{ end }}
+      source: switch
+      switch:
+      - case: 1 # normal
+        set: 
+          source: const
+          value: 0 # % to reset grid loading
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            register:
+              address: 1028 # battery charge current (DC) setpoint, relative
+              type: writemultiple
+              encoding: {{ if (eq .endianness "big") }}float32{{ else }}float32s{{ end }}
+      - case: 2 # hold
+        set:
+          source: const
+          value: 0 # % lock battery
+          set:
+            source: modbus
+            {{- include "modbus" . | indent 10 }}
+            register:
+              address: 1028 # battery charge current (DC) setpoint, relative
```

Comment:
Warum sollte man für hold den Ladestrom setzen? Brauchts hier nicht den Entladestrom?

Reply:
Das Register lässt sich mit dem relativen Wert der Batterieleistung beschreiben.

-100 bis 100%.

Deswegen wird hier 0 geschrieben um sämtliches laden oder entladen zu unterbinden.

Reply:
Deshalb frage ich ja. Warum sollte man bei Entladeschutz (=hold) auch die Ladung verbieten?

Reply:
Das stimmt. Wenn z. B. nur ein Plugin einphasig mit 3.6 kW lädt, ist immer noch ein Überschuss da. Der sollte dann im Speicher landen und nicht im Netz. Kann man das noch anpassen?

Reply:
Warum sollte ich mich als Anwender da entscheiden müssen? Ich will doch Beides?

Reply:
@iseeberg79 Ehrlich gesagt kann ich deinen Ausführungen nicht so ganz folgen, deswegen im folgenden mal meine Einschätzung. 

Zusammengefasst ist ja so, dass `hold` nur mit Register `1042` so umgesetzt werden kann, dass die Batterie auch lädt wenn PV-Strom da ist. 

@andig  erwähnte mal, dass eine Kombination von `batterymode` und `limitsoc` nicht möglich ist. Wenn du aber hier eine Möglichkeit gefunden hast, dann würde ich diese bevorzugen. 

Also für `case 1` und `case 3` Register 1028 verwenden und für `case 2` das Register 1042. 

Ansonsten wäre es für mich auch tragbar, nur das Register 1028 zu verwenden. 

Ich habe die Situation persönlich nie, dass ich trotz Zielladen (hold-Modus) genügend PV Überschuss habe um die Batterie zu laden. Ich wage auch stark zu bezweiflen, dass dieser Usecase bei Anwendern überhaupt eintritt. 

Wenn ich einen Fork für mich machen würde, dann würde ich nur Register 1028 verwenden und darauf pfeifen, dass der WR im hold Modus die Batterie nicht lädt. 

Das ist aber nur meine Meinung und wie ich es umsetzen würde. Dazu müsste halt ein ein Hinweis in die Doku...

Reply:
1030 hat der Plenticore nicht. Das hat nur der Kostal BI. Bin ich auch schon drauf rein gefallen weil ich die Fußnote nicht gelesen habe.